mysqld_exporter.go

```diff
@@ -29,6 +32,10 @@ var (
 		"web.telemetry-path",
 		"Path under which to expose metrics.",
 	).Default("/metrics").String()
+	timeoutOffset = kingpin.Flag(
+		"timeout-offset",
+		"Offset to subtract from timeout in seconds.",
+	).Default("0.5").Float64()
```

Comment:
Should we make this `0.1` if people are going to be scraping with 1s intervals for some collectors?

Reply:
This is why originally I went with percentage of timeout. But I think 0.5 makes more sense. It means, the collector has `0.5s` to stop collecting and send data collected so far to prometheus server. So it's more like a constant that depends on tcp connection and how long it takes for exporter to stop collecting.

But I'm simply ok with any version that works.

Reply:
What do you think @SuperQ ? Should I change it to `0.1`? It's not a big issue for us for now, we are mostly interested in any form of aborting data collection after timeout - so let me know if you would like any changes here.

Reply:
How about we split the difference, make it `0.25`. I think 250ms should be enough cleanup and transmit time for most use cases.

Reply:
Makes perfect sense. Done.